### PR TITLE
Quick start and reference guides

### LINKED ISSUE #5: Write the getting started documentation
The quick-start documentation and user's guide needs to be written up to reach the MVP stage. The docstring for the classes could also use some improvement.

### PR DESCRIPTION
**Pull request by Razoric480**
This PR aims to add a comprehensive API Reference to fix #5 , as well as a lighter weight quick start guide that explains the basics while also providing a full sample class.

I also added a demo that is tied to the quick start guide.

Likely that the quick start guide would become the new README file, but I'll keep them separate for now.

Given we have no guidelines in GDQuest for how to document an API yet, I may have gone overboard with it. But if we're going to be releasing more tools and class-sets, this is probably a discussion we'll be needing to have.

### CODE REVIEW

**Comment by NathanLovato** [comment]
Thanks! I wish GDScript had something like Python, where we can extract docstrings.

When it comes to writing the API, I'd follow the guidelines from another project or company that has great API ref. We can follow the ones from Godot.

Also, I'd write a tool to make it easier to document our APIs, but that may require us to have a programming guideline to differentiate private and virtual functions. So, we'd probably need to use two leading underscores for private stuff, as in Python.

How about the place to write the reference? I'd like to have the equivalent of Python's docstrings in the source code. That way, any dev or user contributing a change will know where to update the function description. A script could then extract these docstrings. There's a problem with that though, it makes the code a pain to edit:

```gdscript
# Caps the agent's movement speed.
var speed_max := 0.0
# Controls the inertia of the agent.
var drag := 1.0
```

Unlike in Python, we don't have a built-in mechanism in GDScript to document the code.

The other efficient option is to have a tool like in the Godot repository, that collects all the functions and dumps them to a JSON file, in which we then write the docs. I could use https://github.com/Scony/godot-gdscript-toolkit to make that quickly, it features a GDScript parser.

File: docs/quickstart.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+You can see the demo in action by opening the `demos/QuickStartDemo.tscn` file in Godot. The result is an agent that approaches the player and hovers near him until it is shot enough times, at which point it will try to flee directly.
+
+More details about how the various steering behaviors function can be found in the [Reference](./reference.md) document.
+
+```ruby
+extends KinematicBody2D
+
+
+export var maximum_speed := 450.0 # Maximum possible linear velocity
+export var maximum_acceleration := 50.0 # Maximum change in linear velocity
```
Comment (NathanLovato) [inline docs/quickstart.md]: I don't know if we discussed it already but I'd recommend having names the other way around: `acceleration_max`, `speed_max`. The most important keyword first. It's the guideline the blender devs use, and as codebases grow, I find it leads to cleaner code. For instance, from the sequencer API:

```py
sequence.frame_start
sequence.frame_end
sequence.frame_duration
sequence.frame_final_start
...
```

File: docs/quickstart.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+export var maximum_rotation_speed := 240 # Maximum rotation velocity represented in degrees
+export var maximum_rotation_accel := 40 # Maximum change in rotation velocity represented in degrees
+export var maximum_health := 100
+export var flee_health_threshold := 20
+
+var velocity := Vector2.ZERO # Represents the current velocity
+var angular_velocity := 0.0 # Represents the current rotational velocity
```
Comment (NathanLovato) [inline docs/quickstart.md]: These comments are not necessary. The name already says explicitly that this is the angular velocity.

File: docs/quickstart.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+var acceleration := GSTTargetAcceleration.new() # Holds the linear and angular components
```
Comment (NathanLovato) [inline docs/quickstart.md]: Use comments consistently. They should always be on the lines before the variable if you're going to have them before functions etc.

**Comment by NathanLovato** [comment]
I just saw the reference: yeah, that was not a good way to go about it. Writing the code reference wasn't part of the task as well: we only talked about a quickstart guide.

For these things, please open a discussion before doing them: you shouldn't write the structure of a code reference by hand in markdown, it makes it hard to update and to change.

File: docs/quickstart.md (hunk 1 above)
Comment (Razoric480) [reply docs/quickstart.md]: We hadn't until now, but I can fold it in throughout the codebase.

Comment (NathanLovato) [reply docs/quickstart.md]: It's more of a detail, but a matter of consistency. Also, at times, it's not clear which word should come first. It's more of a general note.

**Pull request merged**